### PR TITLE
[UI]Added arrow on hovering Collapsed SideNav

### LINKED ISSUE #5123: [UI] The labels don't go away after the collapsed sidebar is opened
#### Current Behavior

The labels on the navigator don't go away when the collapsed sidebar is opened again unless the user refreshes the page.

#### Desired Behavior

The labels should appear only while hovering over the particular list item and go away.
#### Screenshots/Logs

https://user-images.githubusercontent.com/78342516/155689310-ff7e6295-861f-4ae8-99b2-5cfcbc833936.mp4

#### Environment

- Browser:  Firefox
- Host OS: Linux 
- Meshery Server Version: stable-v
- Meshery Client Version: stable-v
- Platform: Docker Kubernetes

---

#### Contributor [Guides](https://docs.meshery.io/project/contributing) and Resources

- 🎨 Wireframes and designs for Meshery UI in [Figma](https://www.figma.com/file/SMP3zxOjZztdOLtgN4dS2W/Meshery-UI)
- 🖥 [Contributing to Meshery UI](https://docs.meshery.io/project/contributing/contributing-ui)
- 🙋🏾🙋🏼 Questions: [Layer5 Discussion Forum](https://discuss.layer5.io) and [Layer5 Community Slack](http://slack.layer5.io)

### PR DESCRIPTION
**Pull request by Harshit-Dandriyal**
Signed-off-by: Harshit-Dandriyal <dandriyalh9318@gmail.com>

**Description**
As discussed in meshey-ui slack channel, I have created hovering arrow on collapsed SideNav and removed lagged labels which sticks to side navbar 

 

https://user-images.githubusercontent.com/56961488/160525728-7041fad8-50dd-4222-8fd0-94e447ed374c.mp4

This PR fixes #5123 

**Notes for Reviewers**

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

**Comment by Harshit-Dandriyal** [comment]
@warunicorn19

**Comment by leecalcote** [comment]
👍

**Comment by asubedy** [comment]
I guess now there is no tooltip text on the parent elements of the nav-bar. This might not be something we want.

**Comment by Harshit-Dandriyal** [comment]
@asubedy but I think there is no need of tooltip text now because whenever we hover over our parent nav, it changes to arrow

**Comment by adithyaakrishna** [comment]
@Harshit-Dandriyal In terms of design, lets consider a new user who isn't accustomed to the Meshery UI, so if he/she collapses the sidebar, it would be difficult for them to determine what those icons mean. So IMHO, labels would be really necessary.

Coming to your discussion regarding the arrow, it would be a nice feature to have along with the tooltip

CC @leecalcote @asubedy @warunicorn19

**Comment by warunicorn19** [comment]
I agree with @adithyaakrishna, the tooltips are necessary.

**Comment by Harshit-Dandriyal** [comment]
@adithyaakrishna Correct, but what should we do then? 
1. Add labels in side of arrows.
2. Add small arrow with listicon(nav item) which shows on hovering and will show labels(may looks conjested).
3. just leave navIcon without arrow as it common sense that it will expand(Already in project).
4.  use arrow on hovering, leave labels which is implemented in this PR.
@leecalcote @warunicorn19 @adithyaakrishna Suggestion please

**Comment by Chadha93** [comment]
@Harshit-Dandriyal I think the simplest answer to all these questions will be - Have the same behavior that we have currently but with the fix that after hover is done tooltip the tooltip should not be there(no re-rendering), It should act as a normal tooltip on hover. example: https://mui.com/components/tooltips/#basic-tooltip

**Comment by adithyaakrishna** [comment]
@Harshit-Dandriyal Any updates on this? 👀

**Comment by Harshit-Dandriyal** [comment]
> @Harshit-Dandriyal Any updates on this? 👀 

I recommended a prototype in meshey-ui, Still waiting for suggestions, If community will approve that, I can continue, Also anyone suggestion is welcome on that prototype :)

**Comment by Harshit-Dandriyal** [comment]
> @Harshit-Dandriyal Do you mind sharing the link to the prototype or figma link here?

Sure :)

https://user-images.githubusercontent.com/56961488/161664638-16db1852-ec63-4b2d-9e60-db5b85c73bd8.mp4

**Comment by Harshit-Dandriyal** [comment]
https://user-images.githubusercontent.com/56961488/162347154-09c3f09e-1d10-450b-86ff-9fc4e7ab150a.mp4

@YashKamboj @adithyaakrishna @asubedy Any Suggestions? :)

**Review by Abhishek-kumar09 — DISMISSED** [review]
I really like this :smiley:  Thanks @Harshit-Dandriyal, great work.

**Comment by asubedy** [comment]
Yes, @Harshit-Dandriyal this looks great. 🚀

**Review by adithyaakrishna — APPROVED** [review]
Looks good to me🎉🎊🎉🎊

**Comment by adithyaakrishna** [comment]
@Harshit-Dandriyal Is there a commit for the change in the video which you uploaded above? 
If not, could you please push it so that we can go ahead and merge it?

**Comment by Harshit-Dandriyal** [comment]
> @Harshit-Dandriyal Is there a commit for the change in the video which you uploaded above? If not, could you please push it so that we can go ahead and merge it?

No, It was just a prototype, I just coded it casually, I was assuring that if a Community like this, I will continue my work on this

**Review by Chadha93 — APPROVED** [review]
@Harshit-Dandriyal this looks good, please do update/push the commit to the pr

File: ui/components/Navigator.js [hunk 1]
```diff
@@ -1091,7 +1098,8 @@ class Navigator extends React.Component {
                   path === href && classes.itemActiveItem
                 )}
                 onClick={() => this.toggleItemCollapse(childId)}
-
+                onMouseOver={() => children && isDrawerCollapsed ? this.setState({ hoveredId : childId }) : null}
+                onMouseLeave={() => !this.state.openItems.includes(childId) ? this.setState({ hoveredId : null }) : null}
```
Comment (Abhishek-kumar09) [inline ui/components/Navigator.js:1102]: What are we doing with these hoverId states??

**Review by Abhishek-kumar09 — APPROVED** [review]
LGTM! The frustating tooltip of drawer are finally going away. Great Work @Harshit-Dandriyal

File: ui/components/Navigator.js (hunk 1 above)
Comment (Harshit-Dandriyal) [reply ui/components/Navigator.js:1102]: They are controlling the state of our friendly arrow on hovering.. :)

**Review by asubedy — APPROVED** [review]
LGTM, great 👍 🚀

**Comment by warunicorn19** [comment]
Looking good, Thank you @Harshit-Dandriyal

**Pull request merged**